interactive_templates/templates/v2/analysis/report_utils.py

```diff
@@ -131,3 +134,113 @@ def calculate_variable_windows_codelist_2(
         ]
 
     return codelist_2_date_range
+
+
+def compute_deciles(measure_table, groupby_col, value_col, has_outer_percentiles=True):
+    """
+    Computes deciles and other percentiles from a measure table.
+
+    Args:
+        measure_table: the measure table to compute the percentiles from
+        groupby_col: the name of the column to group by
+        value_col: the name of the column to compute the percentiles for
+        has_outer_percentiles: whether to compute the nine largest and nine smallest percentiles
+
+    Returns:
+    A dataframe with columns for the grouping column, the value column, and the percentile.
+    """
+    quantiles = np.arange(0.1, 1, 0.1)
+    if has_outer_percentiles:
+        quantiles = np.concatenate(
+            [quantiles, np.arange(0.01, 0.1, 0.01), np.arange(0.91, 1, 0.01)]
+        )
+
+    percentiles = (
+        measure_table.groupby(groupby_col)[value_col]
+        .quantile(pd.Series(quantiles))
+        .reset_index()
+    )
+    percentiles["percentile"] = (percentiles["level_1"] * 100).astype(int)
```

Comment:
I think you will have the repeated outer deciles problem here?

Reply:
Ah yes, good spot, I've added in the fix